src/main/java/io/rpg/controller/PopupController.java

```diff
@@ -51,6 +54,16 @@ public void openPointsPopup(int pointsCount, int x, int y) {
     openTextImagePopup("You earned " + pointsCount + " points!", coinImage, x, y);
   }
 
+  public void openQuestionPopup(Question question, int x, int y, EventHandler<? super MouseEvent> successCallback, EventHandler<? super MouseEvent> failureCallback) {
```

Comment:
EventHandler<? super MouseEvent> is working, but it looks a bit dirty in my opinion. I would use a Runnable or Callable instead and then transformed it to EventHandler<? super MouseEvent> somewhere deeper down the call stack (around viewModel )